albumentations/augmentations/transforms.py

```diff
@@ -2432,24 +2388,18 @@ def apply(
     ) -> np.ndarray:
         return multiply(img, multiplier)
 
-    def get_params_dependent_on_targets(self, params: dict[str, Any]) -> dict[str, Any]:
+    def get_params_dependent_on_data(self, params: dict[str, Any], data: dict[str, Any]) -> dict[str, Any]:
         if self.multiplier[0] == self.multiplier[1]:
             return {"multiplier": self.multiplier[0]}
 
-        img = params["image"]
-
-        num_channels = get_num_channels(img)
+        img_shape = params["shape"]
 
-        shape = img.shape if self.elementwise else [num_channels]
+        shape = img_shape if self.elementwise else [img_shape[-1]]
```

Comment:
We cannot assume that img_shape[-1] is the number of channels.

Use `num_channels = get_num_channels(img)` instead